### PR TITLE
[entropy_src/rtl] add a clean halt/start sequence

### PR DESCRIPTION
**Pull request by mwbranstad**
This change allows entropy_src to stop immediately when disabled.
It also locks all registers when enabled, except for the CONF register.

Signed-off-by: Mark Branstad <mark.branstad@wdc.com>

### CODE REVIEW

File: hw/ip/entropy_src/rtl/entropy_src_main_sm.sv
Comment (tjaychen) [inline hw/ip/entropy_src/rtl/entropy_src_main_sm.sv:108]: this is just a suggestion, completely up to you.
Some of our modules (i believe `otp_ctrl` and `ibex` fall into this category), have an explicit `if (!enable) after the fsm to act as the disable case.  Since these assignments are all blocking, the enable at the end of the fsm will also override everything already done by the FSM.   I don't personally like this style, but just a note if you want to save on some of the `(!enable)` cases.

**Review by tjaychen — APPROVED** [review]
lgtm! once this is in, i will go update the `dif_entropy_smoketest` and put it back into CI!

File: hw/ip/entropy_src/rtl/entropy_src_main_sm.sv
Comment (mwbranstad) [reply hw/ip/entropy_src/rtl/entropy_src_main_sm.sv:108]: Yeah, didn't exactly care for adding all of these statements for each case, but it is an important function, so I guess I will just leave it to make it very obvious within the state.

Comment (mwbranstad) [reply hw/ip/entropy_src/rtl/entropy_src_main_sm.sv:108]: Rethinking this some, I think I have to use this scheme and add some function in some states to "reset" the sha3 block.

Comment (tjaychen) [reply hw/ip/entropy_src/rtl/entropy_src_main_sm.sv:108]: ah yes, if some of the behavior is different per state, then that's definitely true.

**Comment by tjaychen** [comment]
@mwbranstad any objections to me merging this?

**Comment by mwbranstad** [comment]
I think it is ready to go.

**Review by cdgori — COMMENTED** [review]
After the fact, but I did look at this when it was pending review.  It LGTM but I couldn't readily tell if all of the places that needed es_enable bracketing were covered (really hard for me to figure out) - what I saw made sense though.

**Pull request merged**
